### PR TITLE
Custom host changes for VS Scenario

### LINKED ISSUE #3814: Create a custom .NET host which will launch the appropriate func exe, for visual studio use case
### Description

Build a custom .NET host, similar to the .NET isolated placeholder, that uses local.settings.json and the value defined in `FUNCTIONS_WORKER_RUNTIME` to select the appropriate version of the func executable.

The package will include two versions: .NET 8 in-process and .NET 6 in-process func executables. 

Only 64-bit scenarios will be supported, enabling the use of AOT for building the custom host. The func exes should not be self-contained.

### PR DESCRIPTION
**Pull request by aishwaryabh**
### Issue describing the changes in this PR

resolves #3814

This PR creates the custom host needed for VS for the core tools scenario. Today we look at the user's `local.settings.json` to read the values if the function app is dotnet and if the user is running a .NET 8 in-proc app. We then load the appropriate `func.dll` depending on if we want to load an inproc6 or inproc8. The out-of-process scenario will be addressed in a separate PR.
### Pull request checklist

* [ ] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [ ] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [ ] I have added all required tests (Unit tests, E2E tests)

### CODE REVIEW

**Review by mattchenderson — COMMENTED** [review]
I have not yet done any proper review of the actual changeset here, but I have an immediate question regarding the target branch. This should be a separate release that will come out in advance of the OOP host version, right?

File: host/src/FunctionsCustomHost/Logger.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+            if (Configuration.IsTraceLogEnabled)
+            {
+                Log(message);
+            }
+        }
+
+        internal static void Log(string message)
+        {
+            var ts = DateTime.UtcNow.ToString("yyyy-MM-dd HH:mm:ss.fff", CultureInfo.InvariantCulture);
+            Console.WriteLine($"{Configuration.LogPrefix}[{ts}] [FunctionsCustomHost] {message}");
```
Comment (kshyju) [inline host/src/FunctionsCustomHost/Logger.cs:24]: I do not think we should use this log prefix. The log prefix was used in FNH so that the host could read it and log as needed.

File: build/BuildSteps.cs [hunk 2]
```diff
@@ -102,7 +102,7 @@ public static void DotnetPack()
             Shell.Run("dotnet", $"pack {Settings.SrcProjectPath} " +
                                 $"/p:BuildNumber=\"{Settings.BuildNumber}\" " +
                                 $"/p:NoWorkers=\"true\" " +
-                                $"/p:TargetFramework=net6.0 " +  // without TargetFramework, the generated nuspec has incorrect path for the copy files operation.
+                                $"/p:TargetFramework=net8.0 " +  // without TargetFramework, the generated nuspec has incorrect path for the copy files operation.
```
Comment (kshyju) [inline build/BuildSteps.cs:105]: Do we need this change?

File: host/src/FunctionsCustomHost/AppLoader/AppLoader.cs [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    /// <summary>
+    /// Manages loading hostfxr & worker assembly.
+    /// </summary>
+    internal sealed class AppLoader : IDisposable
+    {
+        private IntPtr _hostfxrHandle = IntPtr.Zero;
+        private IntPtr _hostContextHandle = IntPtr.Zero;
+        private bool _disposed;
+
+        internal AppLoader()
```
Comment (kshyju) [inline host/src/FunctionsCustomHost/AppLoader/AppLoader.cs:22]: Remove empty constructor.

File: host/src/FunctionsCustomHost/AppLoader/AppLoader.cs [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+                if (_hostfxrHandle == IntPtr.Zero)
+                {
+                    Logger.Log($"Failed to load hostfxr. hostfxrFullPath:{hostfxrFullPath}");
+                    return -1;
+                }
+
+                Logger.Log($"hostfxr loaded.");
+
+                Logger.Log($"hostfxr initialized with {assemblyPath}");
+                HostFxr.SetAppContextData(_hostContextHandle, "AZURE_FUNCTIONS_NATIVE_HOST", "1");
```
Comment (kshyju) [inline host/src/FunctionsCustomHost/AppLoader/AppLoader.cs:52]: We don't need this line.

File: host/src/FunctionsCustomHost/Configuration/Configuration.cs [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+namespace FunctionsCustomHost
+{
+    /// <summary>
+    /// Represents the application configuration.
+    /// </summary>
+    internal static class Configuration
+    {
+        private const string DefaultLogPrefix = "LanguageWorkerConsoleLog";
```
Comment (kshyju) [inline host/src/FunctionsCustomHost/Configuration/Configuration.cs:11]: See https://github.com/Azure/azure-functions-core-tools/pull/3831/files#r1773819752 We probably can remove this class as it may not be used anywhere.

File: host/src/FunctionsCustomHost/FunctionsCustomHost.csproj [hunk 6]
```diff
@@ -0,0 +1,27 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>net8.0</TargetFramework>
```
Comment (kshyju) [inline host/src/FunctionsCustomHost/FunctionsCustomHost.csproj:5]: I think it is safe to use net9.0 since it is already RC.

File: host/src/FunctionsCustomHost/FunctionsCustomHost.csproj [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+    <DefineConstants Condition=" '$([MSBuild]::IsOSPlatform(`Linux`))' == 'true' ">OS_LINUX</DefineConstants>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <ExportsFile>$(MSBuildThisFileDirectory)exports.def</ExportsFile>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Newtonsoft.Json" Version="13.0.3" />
+	<PackageReference Include="Microsoft.NETCore.DotNetAppHost" Version="8.0.0-preview.4.23259.5" />
```
Comment (kshyju) [inline host/src/FunctionsCustomHost/FunctionsCustomHost.csproj:24]: Use latest version of this package (9x)

File: host/src/FunctionsCustomHost/Program.cs [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+                using var appLoader = new AppLoader();
+
+                var localSettingsJObject = await GetLocalSettingsJsonAsJObjectAsync();
+                var workerRuntime = localSettingsJObject?["Values"]?[EnvironmentVariables.FunctionsWorkerRuntime]?.Value<string>();
+                if (string.IsNullOrEmpty(workerRuntime))
+                {
+                    Logger.Log($"Environment variable '{EnvironmentVariables.FunctionsWorkerRuntime}' is not set.");
+                    return;
+                }
+                if (workerRuntime == DotnetConstants.DotnetWorkerRuntime)
```
Comment (kshyju) [inline host/src/FunctionsCustomHost/Program.cs:27]: Add line break before line 27.

File: host/src/FunctionsCustomHost/Program.cs [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+                Logger.Log($"An error occurred while running FunctionsCustomHost.{exception}");
+            }
+        }
+
+        private static void LoadHostAssembly(AppLoader appLoader, bool isNet8InProc)
+        {
+            var currentDirectory = AppContext.BaseDirectory;
+            var executableName = DotnetConstants.ExecutableName;
+
+            string fileName = "";
```
Comment (kshyju) [inline host/src/FunctionsCustomHost/Program.cs:56]: nit: it's not just name, but the path.

File: host/src/FunctionsCustomHost/Program.cs [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+            fileName = Path.Combine(currentDirectory, isNet8InProc ? DotnetConstants.InProc8DirectoryName: DotnetConstants.InProc6DirectoryName, executableName);
+
+            appLoader.RunApplication(fileName);
+
+            var logMessage = $"FunctionApp assembly loaded successfully. ProcessId:{Environment.ProcessId}";
+            if (OperatingSystem.IsWindows())
```
Comment (kshyju) [inline host/src/FunctionsCustomHost/Program.cs:62]: This is not valid and can be removed.,

File: host/src/FunctionsCustomHost/Program.cs [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+            }
+            catch (Exception exception)
+            {
+                Logger.Log($"An error occurred while running FunctionsCustomHost.{exception}");
+            }
+        }
+
+        private static void LoadHostAssembly(AppLoader appLoader, bool isNet8InProc)
+        {
+            var currentDirectory = AppContext.BaseDirectory;
```
Comment (kshyju) [inline host/src/FunctionsCustomHost/Program.cs:53]: I see we use `Environment.CurrentDirectory` in another place. Let's stick with one and be consistent with that.

File: host/src/FunctionsCustomHost/FunctionsCustomHost.csproj [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+  <PropertyGroup>
+    <DefineConstants Condition=" '$([MSBuild]::IsOSPlatform(`Linux`))' == 'true' ">OS_LINUX</DefineConstants>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <ExportsFile>$(MSBuildThisFileDirectory)exports.def</ExportsFile>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Newtonsoft.Json" Version="13.0.3" />
```
Comment (kshyju) [inline host/src/FunctionsCustomHost/FunctionsCustomHost.csproj:23]: Any reason to not use STJ? FNH uses STJ (So confirmed it is AOT compliant)

File: host/src/FunctionsCustomHost/Program.cs [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+            {
+                logMessage += $", AppPoolId:{Environment.GetEnvironmentVariable(EnvironmentVariables.AppPoolId)}";
+            }
+            Logger.Log(logMessage);
+
+            // Have this here to stall the process
+            Console.ReadLine();
+        }
+
+        private static async Task<JObject> GetLocalSettingsJsonAsJObjectAsync()
```
Comment (kshyju) [inline host/src/FunctionsCustomHost/Program.cs:72]: Could we move this logic' to a different class?

File: host/src/FunctionsCustomHost/FunctionsCustomHost.csproj [hunk 14]
```diff
@@ -0,0 +1,27 @@
...
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>net8.0</TargetFramework>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
+    <AllowUnsafeBlocks>True</AllowUnsafeBlocks>
+    <PublishAot>true</PublishAot>
+    <OptimizationPreference>Speed</OptimizationPreference>
+    <IlcExportUnmanagedEntrypoints>true</IlcExportUnmanagedEntrypoints>
```
Comment (kshyju) [inline host/src/FunctionsCustomHost/FunctionsCustomHost.csproj:11]: @mattchenderson 
What should be the executable name be? Should it be "func" or Can we use another name? I see we have the `localEntryPoint` property value in the feed item, which I am assuming is what VS uses to launch the exe. 

@aishwaryabh  You can set the `AssemblyName` element to change the executable name produced.

File: build/BuildSteps.cs [hunk 15]
```diff
@@ -643,10 +642,22 @@ public static void DotnetPublishForNupkg()
...
+        {
+            // This publishes to the /bin/Release/$targetFramework$/publish
+            Shell.Run("dotnet", $"publish {Settings.CustomHostProjectFile} " +
+                                $"/p:BuildNumber=\"{Settings.BuildNumber}\" " +
+                                $"/p:NoWorkers=\"true\" " +
+                                $"/p:TargetFramework=net8.0 " +
                                 $"/p:CommitHash=\"{Settings.CommitId}\" " +
                                 (string.IsNullOrEmpty(Settings.IntegrationBuildNumber) ? string.Empty : $"/p:IntegrationBuildNumber=\"{Settings.IntegrationBuildNumber}\" ") +
-                                $"-c Release -f net6.0");
+                                $"-c Release -f net8.0 -o {Settings.NupkgPublishDir}");
```
Comment (kshyju) [inline build/BuildSteps.cs:660]: Where will the output of this goes? Don't we need this in the root of the package? We need to make sure we are AOT publishing on the correct platform (not just windows).

Comment (aishwaryabh) [reply build/BuildSteps.cs:660]: so when I was testing locally without specifying the output directory it would go to `host/src/FunctionsCustomHost/bin/Release/net8.0/{framework}/publish`. With the specified output directory it goes to `/bin/Release/{targetFramework}/publish` which is what I thought we wanted

File: host/src/FunctionsCustomHost/FunctionsCustomHost.csproj (hunk 14 above)
Comment (kshyju) [reply host/src/FunctionsCustomHost/FunctionsCustomHost.csproj:11]: Spoke to @mattchenderson  and @fabiocav . The executable name should be `func`.

File: host/src/FunctionsCustomHost/AppLoader/AppLoader.cs [hunk 16]
```diff
@@ -0,0 +1,100 @@
...
+
+                return HostFxr.Run(_hostContextHandle);
```
Comment (kshyju) [inline host/src/FunctionsCustomHost/AppLoader/AppLoader.cs:54]: I think we are missing the code to set `_hostContextHandle`. Looks like Initialize call is missing.

File: host/src/FunctionsCustomHost/Configuration/Configuration.cs (hunk 5 above)
Comment (aishwaryabh) [reply host/src/FunctionsCustomHost/Configuration/Configuration.cs:11]: I think we might still need this class for enabling trace logs though

File: host/src/FunctionsCustomHost/FunctionsCustomHost.csproj (hunk 7 above)
Comment (aishwaryabh) [reply host/src/FunctionsCustomHost/FunctionsCustomHost.csproj:24]: I don't see a 9x version: https://www.nuget.org/packages/Microsoft.NETCore.DotNetAppHost/

File: host/src/FunctionsCustomHost/Program.cs (hunk 11 above)
Comment (aishwaryabh) [reply host/src/FunctionsCustomHost/Program.cs:53]: I think we need AppContext.BaseDirectory though since the FunctionsCustomHost.exe would be in the root (and not the current directory for where the function app is running)

File: host/src/FunctionsCustomHost/FunctionsCustomHost.csproj (hunk 6 above)
Comment (aishwaryabh) [reply host/src/FunctionsCustomHost/FunctionsCustomHost.csproj:5]: I receive this error when trying to do so:
image

File: host/src/FunctionsCustomHost/FunctionsCustomHost.csproj (hunk 14 above)
Comment (aishwaryabh) [reply host/src/FunctionsCustomHost/FunctionsCustomHost.csproj:11]: changed to func. let me know if it looks ok

File: host/src/FunctionsCustomHost/Program.cs [hunk 17]
```diff
@@ -0,0 +1,80 @@
...
+            string filePath = "";
+            filePath = Path.Combine(currentDirectory, isNet8InProc ? DotnetConstants.InProc8DirectoryName: DotnetConstants.InProc6DirectoryName, executableName);
+
+            appLoader.RunApplication(filePath);
+
+            var logMessage = $"FunctionApp assembly loaded successfully. ProcessId:{Environment.ProcessId}";
+            Logger.LogTrace(logMessage);
+
+            // Have this here to stall the process
+            Console.ReadLine();
```
Comment (kshyju) [inline host/src/FunctionsCustomHost/Program.cs:77]: I learned this host will be used only with `host start` command. So that will keep the func.dll/exe running. So, we don't need this line.

File: build/BuildSteps.cs [hunk 18]
```diff
@@ -648,6 +648,18 @@ public static void DotnetPublishForNupkg()
                                 $"-c Release -f net8.0");
         }
 
+        public static void DotnetPublishForCustomHost()
+        {
+            // This publishes to the /bin/Release/$targetFramework$/publish
+            Shell.Run("dotnet", $"publish {Settings.CustomHostProjectFile} " +
+                                $"/p:BuildNumber=\"{Settings.BuildNumber}\" " +
```
Comment (kshyju) [inline build/BuildSteps.cs:655]: I do not think BuildNumber, NoWorkers, IntegrationBuildNumber, CommitHash are used anywhere when building native host.

File: host/src/FunctionsCustomHost/FunctionsCustomHost.csproj (hunk 6 above)
Comment (aishwaryabh) [reply host/src/FunctionsCustomHost/FunctionsCustomHost.csproj]: nvm fixed it

File: host/src/FunctionsCustomHost/Configuration/Configuration.cs (hunk 5 above)
Comment (aishwaryabh) [reply host/src/FunctionsCustomHost/Configuration/Configuration.cs]: nvm removed

File: host/src/FunctionsCustomHost/Program.cs [hunk 19]
```diff
@@ -0,0 +1,109 @@
...
+
+namespace FunctionsCustomHost
+{
+    internal class Program
+    {
+        static async Task Main(string[] args)
+        {
+            try
+            {
+                Logger.Log("Starting FunctionsCustomHost");
```
[the remark this answers is not in the document]
Comment (aishwaryabh) [reply host/src/FunctionsCustomHost/Program.cs]: added args to parse verbose flag

File: host/src/FunctionsCustomHost/AppLoader/AppLoader.cs (hunk 16 above)
Comment (aishwaryabh) [reply host/src/FunctionsCustomHost/AppLoader/AppLoader.cs:68]: added initialize call

File: host/src/FunctionsCustomHost/Program.cs (hunk 17 above)
Comment (aishwaryabh) [reply host/src/FunctionsCustomHost/Program.cs]: removed

File: build/BuildSteps.cs (hunk 15 above)
Comment (aishwaryabh) [reply build/BuildSteps.cs]: removed

File: build-custom-host-artifacts-windows.yml [hunk 20]
```diff
@@ -0,0 +1,47 @@
...
+      workingDirectory: $(Build.SourcesDirectory)/host/src/FunctionsCustomHost
+
+  - task: CopyFiles@2
+    displayName: Copy files
+    inputs:
+      SourceFolder: $(Build.SourcesDirectory)/pkg_output/windows
+      # Publish output will include many other files. We only need func.exe, pdb & nethost.dll
+      Contents: |
+        func.exe
+        func.pdb
```
Comment (kshyju) [inline build-custom-host-artifacts-windows.yml:45]: we don't need the pdb

File: build-custom-host-artifacts-windows.yml [hunk 21]
```diff
@@ -0,0 +1,47 @@
...
+      useGlobalJson: true
+
+  - task: DotnetCoreCLI@2
+    displayName: Dotnet Publish (win-x64)
+    inputs:
+      command: publish
+      publishWebProjects: false
+      zipAfterPublish: false
+      arguments: -c Release -r win-x64 -o $(Build.SourcesDirectory)/pkg_output/windows
+      workingDirectory: $(Build.SourcesDirectory)/host/src/FunctionsCustomHost
```
Comment (kshyju) [inline build-custom-host-artifacts-windows.yml:27]: Instead of calling it FunctionsCustomHost, should we call it "CoreToolsHost" or something along that line?

File: build-custom-host-artifacts-windows.yml [hunk 22]
```diff
@@ -0,0 +1,47 @@
...
+
+  - task: DotnetCoreCLI@2
+    displayName: Dotnet Publish (win-arm64)
+    inputs:
+      command: publish
+      publishWebProjects: false
+      zipAfterPublish: false
+      arguments: -c Release -r win-arm64 -o $(Build.SourcesDirectory)/pkg_output/windows
```
Comment (kshyju) [inline build-custom-host-artifacts-windows.yml:35]: the output folder should be unique for each of these rids. (create subfolder under windows)

File: build-custom-host-artifacts-windows.yml [hunk 23]
```diff
@@ -0,0 +1,47 @@
...
+  - task: CopyFiles@2
+    displayName: Copy files
+    inputs:
+      SourceFolder: $(Build.SourcesDirectory)/pkg_output/windows
+      # Publish output will include many other files. We only need func.exe, pdb & nethost.dll
+      Contents: |
+        func.exe
+        func.pdb
+        nethost.dll
+      TargetFolder: $(Build.ArtifactStagingDirectory)/_functionsCustomHostPackagesWindows
```
Comment (kshyju) [inline build-custom-host-artifacts-windows.yml:47]: Don't we need "PublishPipelineArtifact" task to publish the content  produced from the dotnet publish tasks above ?

File: build/Settings.cs [hunk 24]
```diff
@@ -35,6 +35,8 @@ private static string config(string @default = null, [CallerMemberName] string k
 
         public static readonly string ProjectFile = Path.Combine(SrcProjectPath, "Azure.Functions.Cli.csproj");
 
+        public static readonly string CustomHostProjectFile = Path.GetFullPath("../host/src/FunctionsCustomHost/FunctionsCustomHost.csproj");
```
Comment (kshyju) [inline build/Settings.cs:38]: Is this change needed?

File: host/src/FunctionsCustomHost/FunctionsCustomHost.csproj [hunk 25]
```diff
@@ -0,0 +1,27 @@
...
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>net9.0</TargetFramework>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
+    <AllowUnsafeBlocks>True</AllowUnsafeBlocks>
+    <PublishAot>true</PublishAot>
+    <OptimizationPreference>Speed</OptimizationPreference>
+    <IlcExportUnmanagedEntrypoints>true</IlcExportUnmanagedEntrypoints>
+	<AssemblyName>func</AssemblyName>
```
Comment (kshyju) [inline host/src/FunctionsCustomHost/FunctionsCustomHost.csproj:12]: Fix Indentation.

File: host/src/FunctionsCustomHost/Program.cs [hunk 26]
```diff
@@ -0,0 +1,88 @@
...
+            filePath = Path.Combine(currentDirectory, isNet8InProc ? DotnetConstants.InProc8DirectoryName: DotnetConstants.InProc6DirectoryName, executableName);
+
+            int response = appLoader.RunApplication(filePath, args);
+
+            if (response < 0)
+            {
+                return;
+            }
+
+            var logMessage = $"FunctionApp assembly loaded successfully. ProcessId:{Environment.ProcessId}";
```
Comment (kshyju) [inline host/src/FunctionsCustomHost/Program.cs:84]: Are we loading any function app assembly? Is this log message valid?

Comment (aishwaryabh) [reply host/src/FunctionsCustomHost/Program.cs]: no I fixed the log message

File: build/Settings.cs (hunk 24 above)
Comment (aishwaryabh) [reply build/Settings.cs]: rmeoved

File: build-custom-host-artifacts-windows.yml (hunk 23 above)
Comment (aishwaryabh) [reply build-custom-host-artifacts-windows.yml]: added let me know if it is ok

File: build-custom-host-artifacts-windows.yml (hunk 21 above)
Comment (aishwaryabh) [reply build-custom-host-artifacts-windows.yml]: renamed

File: host/src/CoreToolsHost/Program.cs [hunk 27]
```diff
@@ -0,0 +1,88 @@
...
+namespace CoreToolsHost
+{
+    internal class Program
+    {
+        static bool isVerbose = false;
+        static async Task Main(string[] args)
+        {
+            try
+            {
+                if (args.Contains("--verbose"))
```
Comment (kshyju) [inline host/src/CoreToolsHost/Program.cs:16]: Can simply to ` isVerbose = args.Contains("--verbose");` (Move the string to a constant as well since it's being used in many places.)

File: host/src/CoreToolsHost/Program.cs [hunk 28]
```diff
@@ -0,0 +1,88 @@
...
+                    isVerbose = true;
+                }
+
+                Logger.LogVerbose(isVerbose, "Starting CoreToolsHost");
+
+                using var appLoader = new AppLoader();
+
+                var localSettingsJson = await LocalSettingsJsonParser.GetLocalSettingsJsonAsJObjectAsync();
+                localSettingsJson.RootElement.TryGetProperty("Values", out JsonElement valuesElement);
+                string workerRuntime = "";
```
Comment (kshyju) [inline host/src/CoreToolsHost/Program.cs:27]:
```suggestion
                string workerRuntime = string.Empty;
```

File: host/src/CoreToolsHost/Program.cs [hunk 29]
```diff
@@ -0,0 +1,88 @@
...
+                        Logger.LogVerbose(isVerbose, "Loading inproc8 host");
+                        LoadHostAssembly(appLoader, args, isNet8InProc: true);
+                    }
+                    else
+                    {
+                        // Load host assembly for .NET 6 in proc host
+                        Logger.LogVerbose(isVerbose, "Loading inproc6 host");
+                        LoadHostAssembly(appLoader, args, isNet8InProc: false);
+                    }
+
```
Comment (kshyju) [inline host/src/CoreToolsHost/Program.cs:60]: remove line break

File: host/src/CoreToolsHost/Program.cs [hunk 30]
```diff
@@ -0,0 +1,88 @@
...
+            int response = appLoader.RunApplication(filePath, args);
+
+            if (response < 0)
+            {
+                return;
+            }
+
+            var logMessage = $"Host loaded successfully";
+            Logger.LogVerbose(isVerbose, logMessage);
```
Comment (kshyju) [inline host/src/CoreToolsHost/Program.cs:85]: I don't think this line will execute unless there is an error in loading the assembly.  So may be remove?

File: host/src/CoreToolsHost/Program.cs [hunk 31]
```diff
@@ -0,0 +1,88 @@
...
+        }
+
+        private static void LoadHostAssembly(AppLoader appLoader, string[] args, bool isNet8InProc)
+        {
+            var currentDirectory = AppContext.BaseDirectory;
+            var executableName = DotnetConstants.ExecutableName;
+
+            string filePath = "";
+            filePath = Path.Combine(currentDirectory, isNet8InProc ? DotnetConstants.InProc8DirectoryName: DotnetConstants.InProc6DirectoryName, executableName);
+
```
Comment (kshyju) [inline host/src/CoreToolsHost/Program.cs:76]: Verbose log the filePath.

**Review by kshyju — APPROVED** [review]

File: host/src/CoreToolsHost/Program.cs (hunk 30 above)
Comment (aishwaryabh) [reply host/src/CoreToolsHost/Program.cs]: rmeoved

File: host/src/CoreToolsHost/LocalSettingsJsonParser.cs [hunk 32]
```diff
@@ -0,0 +1,34 @@
...
+            {
+                string fileContent = "";
+                using (var fileStream = File.Open(fullPath, FileMode.Open, FileAccess.Read, FileShare.ReadWrite | FileShare.Delete))
+                using (var streamReader = new StreamReader(fileStream))
+                {
+                    fileContent = await streamReader.ReadToEndAsync();
+                }
+                if (!string.IsNullOrEmpty(fileContent))
+                {
+                    var localSettingsJObject = JsonDocument.Parse(fileContent);
```
Comment (drdamour) [inline host/src/CoreToolsHost/LocalSettingsJsonParser.cs:26]: this line gave me a bad weekend :(  glad #3866 is addressing some of it (but still not allowing my sweet sweet trailing commas :D
